pkg/ebpf/telemetry/c/ebpf_instrumentation.c

```diff
@@ -0,0 +1,22 @@
+#include "bpf_telemetry.h"
+#include "bpf_helpers.h"
+#include "map-defs.h"
+#include "compiler.h"
+
+// ebpf_instrumentation__trampoline_handler is the target for the trampoline jump
+// This program caches a pointer to the telemetry map on the stack at offset 512
+SEC("ebpf_instrumentation/trampoline_handler")
+int ebpf_instrumentation__trampoline_handler() {
+    u64 key = 0;
+    instrumentation_blob_t* tb = bpf_map_lookup_elem(&bpf_instrumentation_map, &key);
+    if (tb == NULL) {
+        asm ("r2 = 0");
+        // we need to set this stack slot to avoid verifier error on access
+        asm("*(u64 *)(r10 - 512) = r2");
```

Comment:
Can you use an immediate store instead?
```suggestion
        // we need to set this stack slot to avoid verifier error on access
        asm("*(u64 *)(r10 - 512) = 0");
```

Reply:
If not, please add a comment why, because using the register is non-obvious.

Reply:
The compiler complains of invalid assembly. Will try to see what the expected syntax is for storing an immediate to a memory location.

Reply:
weird, I found this on https://www.kernel.org/doc/html/v5.17/bpf/instruction-set.html
`*(size *) (dst_reg + off) = imm32`